cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/CapacityConfiguration.java

```diff
@@ -0,0 +1,257 @@
+
+    private final TreeMap<Integer, CapacityEntry> capacityEntries;
+
+    /**
+     * Constructor
+     *
+     * @param reconciliation        Reconciliation marker.
+     * @param spec                  Spec of the Kafka custom resource.
+     * @param kafkaBrokerNodes      List of the broker nodes which are part of the Kafka cluster.
+     * @param kafkaStorage          A map with storage configuration used by the Kafka cluster and its node pools.
+     * @param kafkaBrokerResources  A map with resource configuration used by the Kafka cluster and its broker pools.
+     */
+    public CapacityConfiguration(
+            Reconciliation reconciliation,
+            KafkaSpec spec,
+            Set<NodeRef> kafkaBrokerNodes,
+            Map<String, Storage> kafkaStorage,
+            Map<String, ResourceRequirements> kafkaBrokerResources
+    ) {
+        this.capacityEntries = generateCapacityEntries(reconciliation, spec.getCruiseControl(), kafkaBrokerNodes,
+                kafkaStorage, kafkaBrokerResources);
+    }
+
+    private static Map<Integer, BrokerCapacityOverride> processBrokerCapacityOverrides(Reconciliation reconciliation,
+                                                                                       Set<NodeRef> kafkaBrokerNodes,
+                                                                                       BrokerCapacity brokerCapacity) {
+        Map<Integer, BrokerCapacityOverride> overrideMap = new HashMap<>();
+
+        if (brokerCapacity != null && brokerCapacity.getOverrides() != null && !brokerCapacity.getOverrides().isEmpty()) {
+            for (BrokerCapacityOverride override : brokerCapacity.getOverrides()) {
+                List<Integer> ids = override.getBrokers();
+
+                for (int id : ids) {
+                    if (overrideMap.containsKey(id)) {
+                        LOGGER.warnCr(reconciliation, "Duplicate broker id {} found in overrides, using first occurrence.", id);
+                    } else if (kafkaBrokerNodes.stream().noneMatch(node -> node.nodeId() == id)) {
+                        LOGGER.warnCr(reconciliation, "Ignoring broker capacity override for unknown node ID {}", id);
+                    } else {
+                        overrideMap.put(id, override);
+                    }
+                }
+            }
+        }
+
+        return overrideMap;
+    }
+
+    private static TreeMap<Integer, CapacityEntry> generateCapacityEntries(Reconciliation reconciliation,
+                                         CruiseControlSpec spec,
+                                         Set<NodeRef> kafkaBrokerNodes,
+                                         Map<String, Storage> kafkaStorage,
+                                         Map<String, ResourceRequirements> kafkaBrokerResources) {
+        TreeMap<Integer, CapacityEntry> capacityEntries = new TreeMap<>();
+        BrokerCapacity generalBrokerCapacity = spec.getBrokerCapacity();
```

Comment:
maybe commonBrokerCapacity? But change only if you like the suggestion otherwise just ignore me ;-)